packages/picasso.js/src/core/dock-layout/dock-layout.js

```diff
@@ -95,6 +95,67 @@ function reduceSingleLayoutRect(logicalContainerRect, reducedRect, edgeBleed, c,
   return true;
 }
 
+
+/**
+ * Updates the visible and hidden components based on components that are docked to other components.
+ * For example, assume a component called myRect:
+ * {
+ *  key: 'myRect',
+ *  type: 'rect',
+ *  dock: 'bottom'
+ * }
+ * and a component called myLine:
+ * {
+ *  key: 'myLine',
+ *  type: 'line',
+ *  dock: '@myRect'
+ * }
+ * if the layout engine decides to hide myRect, then myLine should be hidden as well.
+ * @param {Array} components - Components to be decided if they should be hidden or not.
+ * @param {Array} hiddenComponents - Components that are already hidden.
+ * @returns {Object} containing the new visible components and additional components to be hidden.
+ * @ignore
+ */
+function updateForComponentsDockedAtRemovedComponents(components, hiddenComponents) {
+  const visibleComponents = components.slice();
+
+  if (hiddenComponents.length === 0) {
+    return { visibleComponents, dockedAtRemovedComponents: [] };
+  }
+
+  const dockedAtRemovedComponents = [];
+
+  components.forEach((comp, idx) => {
+    let nbrOfHidden = 0;
+    const dock = comp.config && comp.config.dock();
+    const referencedDocks = dock.split(',').map(s => s.replace(' ', ''));
+    const newTargetComponents = referencedDocks.slice();
+
+    // For multiple @ references
+    referencedDocks.forEach((referencedDock, i) => {
+      const found = hiddenComponents.some((hiddenComp) => {
+        const key = hiddenComp.ctx && hiddenComp.ctx.key;
+        return referencedDock === `@${key}`;
+      });
+
+      if (found) {
+        newTargetComponents.splice(i, 1);
+        nbrOfHidden++;
+      }
+    });
+
+    // Hide the component only if all the referenced components are hidden
+    if (nbrOfHidden === referencedDocks.length) {
+      visibleComponents.splice(idx, 1);
+      dockedAtRemovedComponents.push(comp.instance);
+    } else if (nbrOfHidden > 0) {
+      // Reassigning new dock target(s) if more that one of the referenced components are visible
+      comp.config.dock = () => newTargetComponents.join();
```

Comment:
`config.dock()` takes an optional parameter to re-assign the dock value. However, you cannot re-assign it at all, because you don't know if the user definition for the `dock` setting was a function or not.

Reply:
Removed